include/boost/random/detail/uniform_int_float.hpp

```diff
@@ -60,7 +60,7 @@ class uniform_int_float
     result_type operator()()
     {
         base_result range = static_cast<base_result>((max)())+1;
-        return static_cast<result_type>(_rng() * range);
+        return static_cast<result_type>(static_cast<base_result>(_rng() * range));
```

Comment:
> I'm somewhat curious why the expression templates don't support these 
> conversions, when multiprecision::number itself does. I do notice that 
> detail::expression has an (explicit) conversion to bool.
>
Good question, it got overlooked I guess, conversions between number<> 
types are handled via converting constructors it's just conversions to 
other types (built in ones for example) that's missing.  I realise this 
would simply those horrible double-casts, let me experiment and see if I 
can add support for this to multiprecision without breaking anything.